src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoMending.java

```diff
@@ -0,0 +1,289 @@
+        .build()
+    );
+
+    private int tickDelayLeft;
+
+    public AutoMending() {
+        super(Categories.Player, "auto-mending", "Automatically mends items in your offhand using XP bottles.");
+    }
+
+    @Override
+    public void onActivate() {
+        tickDelayLeft = tickDelay.get();
+    }
+
+    @EventHandler
+    private void onTick(TickEvent.Pre event) {
+        if (tickDelayLeft <= 0) {
+            tickDelayLeft = tickDelay.get();
+
+            if (mendElytra.get() && mendItem(findItemInInventory(Items.ELYTRA), elytraThreshold.get(), elytraMaxDurability.get())) return;
+            if (mendChestplate.get() && mendItem(findItemInInventory(Items.IRON_CHESTPLATE, Items.DIAMOND_CHESTPLATE, Items.NETHERITE_CHESTPLATE), chestplateThreshold.get(), chestplateMaxDurability.get())) return;
+            if (mendLeggings.get() && mendItem(findItemInInventory(Items.IRON_LEGGINGS, Items.DIAMOND_LEGGINGS, Items.NETHERITE_LEGGINGS), leggingsThreshold.get(), leggingsMaxDurability.get())) return;
+            if (mendBoots.get() && mendItem(findItemInInventory(Items.IRON_BOOTS, Items.DIAMOND_BOOTS, Items.NETHERITE_BOOTS), bootsThreshold.get(), bootsMaxDurability.get())) return;
+            if (mendHelmet.get() && mendItem(findItemInInventory(Items.IRON_HELMET, Items.DIAMOND_HELMET, Items.NETHERITE_HELMET), helmetThreshold.get(), helmetMaxDurability.get())) return;
+            if (mendSword.get() && mendItem(findItemInInventory(Items.IRON_SWORD, Items.DIAMOND_SWORD, Items.NETHERITE_SWORD), swordThreshold.get(), swordMaxDurability.get())) return;
+            if (mendPickaxe.get() && mendItem(findItemInInventory(Items.IRON_PICKAXE, Items.DIAMOND_PICKAXE, Items.NETHERITE_PICKAXE), pickaxeThreshold.get(), pickaxeMaxDurability.get())) return;
+        } else {
+            tickDelayLeft--;
+        }
+    }
+
+    private ItemStack findItemInInventory(Item... items) {
+        for (Item item : items) {
+            for (ItemStack stack : mc.player.getInventory().main) {
+                if (stack.getItem() == item) return stack;
+            }
+            for (ItemStack stack : mc.player.getInventory().armor) {
+                if (stack.getItem() == item) return stack;
+            }
+        }
+        return null;
+    }
+
+    private boolean mendItem(ItemStack stack, int threshold, int maxDurability) {
+        if (stack == null) return false;
+
+        int currentDurability = stack.getMaxDamage() - stack.getDamage();
+        int percentage = (currentDurability * 100) / stack.getMaxDamage();
+
+        if (percentage <= threshold && percentage < maxDurability) {
+            int slot = mc.player.getInventory().indexOf(stack);
+            if (slot == -1) return false;
+
+            InvUtils.move().from(slot).to(SlotUtils.OFFHAND);
+            throwXPBottles();
+            InvUtils.move().from(SlotUtils.OFFHAND).to(slot);
+            return true;
+        }
+        return false;
+    }
+
+    private void throwXPBottles() {
```

Comment:
please rewrite this method so that the rotations and swapping are not done on every iteration of the while loop